Lib/asyncio/proactor_events.py

```diff
@@ -232,8 +232,9 @@ class _ProactorBaseWritePipeTransport(_ProactorBasePipeTransport,
 
     def write(self, data):
         if not isinstance(data, (bytes, bytearray, memoryview)):
-            raise TypeError('data argument must be byte-ish (%r)',
-                            type(data))
+            msg = ("data argument must be a bytes-like object, not '%s'" %
```

Comment:
@berkerpeksag AFAIK `repr`ing in error messages is used to play it safe in cases where the object happens to be an instance of `bytes`; the `__name__` special attribute is always a string so `%s` is always certain to work. Apart from that, I followed along with other cases of formatting `__name__` I encountered in other error messages. 

@AraHaan The author of the module prefers using old-style formatting, so I stuck with that to keep it consistent. :-)